### PR TITLE
fix: Update block-dynamic-connection to use JSON toolbox (#1238)

### LINKED ISSUE #1238: Update block-dynamic-connection to use JSON toolbox definitions instead of xml
**Category**
 - Plugins

**Component**

block-dynamic-connection

**Is your feature request related to a problem? Please describe.**

Many of the examples and plugins use XML toolboxes instead of the new JSON API. We should have our example code use the new APIs to best model what other developers can do with Blockly.

**Describe the solution you'd like**

Update the code in blockly-samples to use the JSON toolbox API.

**Additional context**

To work on this issue, first read the [background information](https://developers.google.com/blockly/guides/contribute/samples) about how to contribute to samples and verify your changes.

1. Learn about [defining a toolbox](https://developers.google.com/blockly/guides/configure/web/toolbox) and read about the XML format vs the JSON format.
2. In blockly-samples, navigate to the block-dynamic-connection plugin's directory and run `npm run start` to start the playground. Look at the toolbox and what blocks are available.
3. Write a new copy of the toolbox definition in the JSON format instead of the XML format. The toolbox definition in most cases should now be in `test/index.js` instead of `test/index.html`.
4. Refresh the playground page and make sure the toolbox looks the same as it did before. You can compare against the version in master on the [samples site](https://google.github.io/blockly-samples/) to make sure the behavior hasn't changed.
5. Follow the rest of the instructions in the Contributing docs to verify your change and open a PR.

### PR DESCRIPTION
**Pull request by uyamini**
fixes #1238

### CODE REVIEW

**Comment by debaraj-barua** [comment]
@uyamini Perhaps you can rename the PR to something like `fix: Update block-dynamic-connection to use JSON toolbox` if you want to add the issue number, you should add a `#` before the number eg: `fix: Update block-dynamic-connection to use JSON toolbox (#1238)`

File: plugins/block-dynamic-connection/test/index.js
Comment (debaraj-barua) [inline plugins/block-dynamic-connection/test/index.js]: @uyamini I guess you should also remove the XML block from the `index.html`. But maybe one of the maintainers can clarify on that. 
  
 Additionally, You should add a comma after the toolbox.

Comment (BeksOmega) [reply plugins/block-dynamic-connection/test/index.js]: Yes! Removing the XML is much appreciated =)

**Comment by BeksOmega** [comment]
Looks like there's still a small lint problem, you might be able to do `npm run lint:fix` to fix it. Or you can definitely run `npm run lint` to see what the issue is =)

**Review by BeksOmega — APPROVED** [review]
LGTM! (looks good to me) Thank you for your contribution @uyamini !

I'll just wait for the CI (continuous integration) to pass, and then get this merged!

**Comment by rachel-fenichel** [comment]
@uyamini Thanks for fixing this issue!

**Pull request merged**
